rewrite-core/src/main/java/org/openrewrite/SourceFileWithReferences.java

```diff
@@ -18,34 +18,48 @@
 import lombok.AccessLevel;
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.NonFinal;
 import org.jspecify.annotations.Nullable;
 import org.openrewrite.trait.Reference;
 
+import java.lang.ref.SoftReference;
 import java.util.*;
 
 @Incubating(since = "8.39.0")
-public interface SourceFileWithReferences extends SourceFile {
+public abstract class SourceFileWithReferences implements SourceFile {
```

Comment:
I wonder if this type of change to tree package classes might trip up the deserialization we do for stored LSTs. 🤔

Reply:
O boy, I don't know about this. Could you figure this out or should we just revert the changes, for safety reasons?

Reply:
I will kinda revert it, but still do some enhancement. As I was thinking about a different way to write this, but keep the interface intact, I saw that the `SourceFile` in Reference does not make a lot of sense. So by dropping that field, things become cleaner as well.